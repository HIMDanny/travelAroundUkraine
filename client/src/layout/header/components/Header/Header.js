/* eslint-disable  react/prop-types */
import React, { useState, useEffect } from 'react';
import { AppBar, Toolbar, Container, useScrollTrigger, Slide } from '@mui/material';
import { BurgerMenu, TabletMenu, LogoHeader, HeaderSearchField } from '..';
import { LogOutModal } from '../../../../components';

const HideOnScroll = ({ children }) => {
  const trigger = useScrollTrigger();
  return (
    <Slide appear={false} direction="down" in={!trigger}>
      {children}
    </Slide>
  );
};

const Header = () => {
  const [scrollY, setScrollY] = useState();

  useEffect(() => {
    const handleScrollY = () => setScrollY(window.scrollY);
    document.addEventListener('scroll', handleScrollY);

    return () => document.removeEventListener('scroll', handleScrollY);
  }, []);

  const appBarSX = {
    background: 'primary.main',
    boxShadow: scrollY > 400 ? null : 'none',
    py: { xs: 1, tablet: 1.5, laptop: 2 },
    opacity: 0.95,
    transition: '0.3s !important',
  };

  return (
    <HideOnScroll>
      <AppBar sx={appBarSX}>
        <Container>
          <Toolbar disableGutters>
            <LogoHeader />
            <HeaderSearchField />
            <BurgerMenu />
            <TabletMenu />
            <LogOutModal />
          </Toolbar>
        </Container>
      </AppBar>
    </HideOnScroll>
  );
};

export default Header;
