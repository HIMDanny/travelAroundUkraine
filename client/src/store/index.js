import { configureStore } from '@reduxjs/toolkit';
import userReducer from './slices/userSlice/userSlice';
import tourReducer from './slices/tourSlice/tourSlice';
import catalogueReducer from './slices/catalogueSlice/catalogueSlice';
import cartReducer from './slices/cartSlice/cartSlice';
import orderSlice from './slices/orderSlice/orderSlice';
import inFavoritesReducer from './slices/inFavoritesSlice/inFavoritesSlice';

const store = configureStore({
  reducer: {
    userReducer,
    tour: tourReducer,
    catalogue: catalogueReducer,
    cart: cartReducer,
    order: orderSlice,
    favorites: inFavoritesReducer,
  },
});

export default store;
