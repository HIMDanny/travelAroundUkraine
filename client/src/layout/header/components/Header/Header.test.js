import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../../../store';
import Header from './Header';

jest.mock('@mui/material/AppBar', () => ({ children }) => <header>{children}</header>);
jest.mock('@mui/material/Container', () => ({ children }) => <div>{children}</div>);
jest.mock('@mui/material/Toolbar', () => ({ children }) => <div>{children}</div>);
jest.mock('@mui/material/Slide', () => ({ children }) => <div>{children}</div>);
jest.mock('../../components/BurgerMenu/BurgerMenu.js', () => () => <div>BurgerMenu</div>);
jest.mock('../../components/TabletMenu/TabletMenu.js', () => () => <div>TabletMenu</div>);
jest.mock('../../components/HeaderSearchField/HeaderSearchField.js', () => () => <div>HeaderSearchField</div>);

describe('Render Header', () => {
  test('should Header render', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <BrowserRouter>
          <Header />
        </BrowserRouter>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
