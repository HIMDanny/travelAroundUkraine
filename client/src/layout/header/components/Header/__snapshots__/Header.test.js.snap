// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Header should Header render 1`] = `
<DocumentFragment>
  <div>
    <header>
      <div>
        <div>
          <div>
            LogoHeader
          </div>
          <div>
            HeaderSearchField
          </div>
          <div>
            BurgerMenu
          </div>
          <div>
            TabletMenu
          </div>
        </div>
      </div>
    </header>
  </div>
</DocumentFragment>
`;
